projects/packages/publicize/src/rest-api/class-connections-controller.php

```diff
@@ -0,0 +1,434 @@
+				'enum'        => array(
+					'ok',
+					'broken',
+				),
+			),
+			'test_success' => array(
+				'description' => __( 'Did the Jetpack Social connection test pass?', 'jetpack-publicize-pkg' ),
+				'type'        => 'boolean',
+			),
+			'error_code'   => array(
+				'description' => __( 'Jetpack Social connection error code', 'jetpack-publicize-pkg' ),
+				'type'        => 'string',
+			),
+			'test_message' => array(
+				'description' => __( 'Jetpack Social connection success or error message', 'jetpack-publicize-pkg' ),
+				'type'        => 'string',
+			),
+			'can_refresh'  => array(
+				'description' => __( 'Can the current user refresh the Jetpack Social connection?', 'jetpack-publicize-pkg' ),
+				'type'        => 'boolean',
+			),
+			'refresh_text' => array(
+				'description' => __( 'Message instructing the user to refresh their Connection to the Jetpack Social service', 'jetpack-publicize-pkg' ),
+				'type'        => 'string',
+			),
+			'refresh_url'  => array(
+				'description' => __( 'URL for refreshing the Connection to the Jetpack Social service', 'jetpack-publicize-pkg' ),
+				'type'        => 'string',
+				'format'      => 'uri',
+			),
+		);
+
+		return array_merge(
+			$deprecated_fields,
+			$connection_fields,
+			$test_fields
+		);
+	}
+
+	/**
+	 * Schema for the endpoint.
+	 *
+	 * @return array
+	 */
+	public function get_item_schema() {
+		if ( $this->schema ) {
+			return $this->add_additional_fields_schema( $this->schema );
+		}
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'jetpack-publicize-connection',
+			'type'       => 'object',
+			'properties' => $this->get_connection_schema_properties(),
+		);
+
+		$this->schema = $schema;
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Helper for retrieving Connections. Used by this endpoint and by
+	 * the Connection Test Result endpoint.
+	 *
+	 * @internal
+	 * @param bool $run_tests Whether to run the tests to check the health of the connections.
+	 * @return array
+	 */
+	protected static function get_connections( $run_tests = false ) {
+		global $publicize;
+
+		$items        = array();
+		$test_results = $run_tests ? self::get_connections_test_status() : array();
+
+		foreach ( (array) $publicize->get_services( 'connected' ) as $service_name => $connections ) {
+			foreach ( $connections as $connection ) {
+				$connection_meta = $publicize->get_connection_meta( $connection );
+				$connection_data = $connection_meta['connection_data'];
+				$connection_id   = $publicize->get_connection_id( $connection );
+				$username        = $publicize->get_username( $service_name, $connection );
+				$item            = array_merge(
+					array(
+						'connection_id'        => $connection_id,
+						'display_name'         => $publicize->get_display_name( $service_name, $connection ),
+						'external_handle'      => $username,
```

Comment:
`external_handle` handle is not actually the username. It's a little different. With this PR, I get this, which is incorrect:

<img width="292" alt="Screenshot 2024-12-18 at 9 31 15 AM" src="https://github.com/user-attachments/assets/4eef957a-7d01-4e59-88a5-9ff908a8c790" />

You can see the `get_external_handle` method I added in #40607

Reply:
Yes it seems strange. Why does https://github.com/Automattic/jetpack/pull/40607/files#diff-540cd16120af78cc4afc265ca572d7dac4674ebd81ec12f9479bf079929dc04aR612-R629?

Reply:
Because we don't have it for Facebook pages for example. We can surely improve on that to ensure that our keyring service has all that information.